From ea8a4b37b3f89680837b77bf9f2f910fc147bd48 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Tue, 6 Aug 2024 23:53:39 +0200
Subject: [PATCH 3/4] winegstreamer: Also flush token queue on seek.

This is needed by video_frame_sink in mfmediaengine to properly restart playback after pausing if the unpausing
operation is a seek.
---
 dlls/winegstreamer/media_source.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winegstreamer/media_source.c b/dlls/winegstreamer/media_source.c
index 030d0c1b9a2..6e02dc9a723 100644
--- a/dlls/winegstreamer/media_source.c
+++ b/dlls/winegstreamer/media_source.c
@@ -658,7 +658,7 @@ static HRESULT media_source_start(struct media_source *source, IMFPresentationDe
                 AM_SEEKING_AbsolutePositioning, AM_SEEKING_NoPositioning);
 
     for (i = 0; i < source->stream_count; i++)
-        flush_token_queue(source->streams[i], position->vt == VT_EMPTY);
+        flush_token_queue(source->streams[i], TRUE);
 
     return IMFMediaEventQueue_QueueEventParamVar(source->event_queue,
             seek_message ? MESourceSeeked : MESourceStarted, &GUID_NULL, S_OK, position);
-- 
2.46.0

